parser/mysql/privs.go

```diff
@@ -314,6 +314,9 @@ var AllDBPrivs = Privileges{SelectPriv, InsertPriv, UpdatePriv, DeletePriv, Crea
 // AllTablePrivs is all the privileges in table scope.
 var AllTablePrivs = Privileges{SelectPriv, InsertPriv, UpdatePriv, DeletePriv, CreatePriv, DropPriv, IndexPriv, ReferencesPriv, AlterPriv, CreateViewPriv, ShowViewPriv}
 
+// AllowedTablePrivs is all the privileges allowed in table scope.
+var AllowedTablePrivs = Privileges{SelectPriv, InsertPriv, UpdatePriv, DeletePriv, CreatePriv, DropPriv, IndexPriv, ReferencesPriv, AlterPriv, CreateViewPriv, ShowViewPriv, GrantPriv, UsagePriv, AllPriv}
```

Comment:
I don't think this should be added.
By adding this, you are changing the behaviour of `GRANT GRANT OPTION ON NonExistingDB.NonExistingTable TO testuser` and `GRANT USAGE ON NonExistingDb.NonExistingTabler TO testuser`. In MySQL the behaviour of these two grants are not documented to my knowledge, but they do return Error 1146, table doesn't exist.

I think it is better to keep the special handling of privileges not included in AllTablePrivs in this case, since it is only done in a single place.

Reply:
No, the behavior doesn't change, it is just used to look up table level allowed privileges, to return `ErrIllegalGrantForTable` before table lookup. These two privileges are allowed to be granted to table, so I'd like to include these here.
```
mysql> GRANT GRANT OPTION ON NonExistingDB.NonExistingTable TO testuser;
ERROR 1146 (42S02): Table 'NonExistingDB.NonExistingTable' doesn't exist
mysql> GRANT USAGE ON NonExistingDb.NonExistingTabler TO testuser;
ERROR 1146 (42S02): Table 'NonExistingDb.NonExistingTabler' doesn't exist
```
If you would like I will write test case to ensure that.

Reply:
Update: this has been removed, I think it can be marked as resolved.